### PR TITLE
Major Code Cleanup

### PR DESCRIPTION
**Pull request by kg185155**
## Description

This is a general code cleanup but the main changes include:

1. Changing the folder name from `/lib/hooks/` to `/lib`/swr` as the files weren't hooks at all
2. Adding a default `<Layout />` component as all of the pages on the "public" side of the site were the same and there was a lot of duplicate code
3. Putting the categories menu into the header instead of fetching on each individual header. Also putting this into an `swr` call.
4. Adding a github action to lint all the files
5. Splitting up the `isError`, `isLoading`, and success state into three separate `if` statements instead of doing it all inline

### CODE REVIEW

**Review by co185063 — APPROVED** [review]
LGTM!

File: components/public/home/HomeMap.js [hunk 1]
```diff
@@ -1,46 +1,58 @@
...
+      <div className="my-5">
+        <small className="text-muted center">
+          {`Uhoh, we've hit an error.`}
+          <code>{JSON.stringify(isError.info)}</code>
+        </small>
+      </div>
+    );
+  }
+
+  console.log('the data', data.response.data.pageContent);
```
Comment (co185063) [inline components/public/home/HomeMap.js:27]: Console.log()

File: pages/catalog.js [hunk 2]
```diff
@@ -1,59 +1,62 @@
 import { useContext } from 'react';
-import Head from 'next/head';
-import Header from '~/components/public/Header';
-import Footer from '~/components/public/Footer';
+import { Row, Col } from 'reactstrap';
+
```
Comment (co185063) [inline pages/catalog.js:3]: extra line

**Review by co185063 — CHANGES_REQUESTED** [review]
I'm going to assume everything works well on your local but I would take this time to double and triple check going through the entire site to make sure nothing is broken. Otherwise please make these small changes I left.

**Comment by kg185155** [comment]
@tm185221 @co185063 I found a couple issues and have fixed them in the latest changes. I also made some visual changes in different parts of the site. If you dont mind giving it a QA locally and making sure it all works, that would be greatly appreciated. Want to make sure we are all comfortable with the release

**Review by co185063 — COMMENTED** [review]
So the only problem that I seem to have is with the map and this console error regarding the map. I would imagine this is an issue with a key though and if it works on your local with an appropriate key then it should be good, but I will give my approval only after you check if the map is working on your local.
image
Capture

**Review by co185063 — APPROVED** [review]

**Pull request merged**